core/validation/blockValidator.go

```diff
@@ -19,7 +19,6 @@ func VerifyBlock(block *ledger.Block, ld *ledger.Ledger, completely bool) error
 
 	flag, err := VerifySignableData(block)
 	if flag && err == nil {
-		return nil
```

Comment:
if the "return nil" be removed, the if .. else just have no meaning here, so just adjust the code to the if condition like below 

if  flag == false  || err != nil {
    return err 
}